Gauge/Measurement.hs

```diff
@@ -184,30 +233,62 @@ getGCStatistics = do
   \(_::Exc.SomeException) -> return Nothing
 #endif
 
+#ifdef GAUGE_MEASURE_TIME_NEW
```

Comment:
This may not be necessary. We can just have the new code and remove the old one. Will be easier to maintain. I do not see much risk.

Reply:
I don't plan to keep this long I hope, but I would like to convinced myself that there's nothing wrong with the new measurement code before switching completely, and having the ability to quickly compared the number is paramount. Once #16 in on the way, it's likely that I'll drop the old code to prevent to have to maintain the old code path.